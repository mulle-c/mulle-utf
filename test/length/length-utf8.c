#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>

static char  *text_with_emoji = "\x3c\xf0\x9f\x8e\xb2\x3e";
static char  *text_with_euro = "\x41\xe2\x82\xac\x73\x69\x67\x6e";

static void   test( mulle_utf8_t *s)
{
   printf( "utf8  length: %zu\n", mulle_utf8_strlen( s));
   printf( "utf16 length: %zu\n", mulle_utf8_utf16length( s, -1));
   printf( "utf32 length: %zu\n", mulle_utf8_utf32length( s, -1));
}


int  main()
{
   test( text_with_emoji);
   // surrogates are not valid in UTF8 though
   test( text_with_euro);
   return( 0);
}

