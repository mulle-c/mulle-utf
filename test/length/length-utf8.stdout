utf8  length: 6
utf16 length: 4
utf32 length: 3
utf8  length: 8
utf16 length: 6
utf32 length: 6
